cg/meta/deliver_ticket.py

```diff
@@ -22,108 +22,138 @@ def __init__(self, config: CGConfig):
+        cases: List[models.Family] = self.get_all_cases_from_ticket(ticket_id=ticket_id)
         date = str(cases[0].ordered_at)
         split_date = date.split(" ")
         return split_date[0]
 
+    def generate_output_filename(
+        self, dir_path: Path, date: str, dir_name: Path, read_direction: int
+    ) -> Path:
+        fastq_file_name = str(os.path.basename(dir_name)) + "_" + str(read_direction) + ".fastq.gz"
+        if date:
+            fastq_file_name = (
+                date
+                + "_"
+                + str(os.path.basename(dir_name))
+                + "_"
+                + str(read_direction)
+                + ".fastq.gz"
+            )
+        output = dir_path / fastq_file_name
+        return output
+
+    def get_current_read_direction(self, dir_path: Path, read_direction: int) -> list:
+        same_direction = []
+        for file in os.listdir(dir_path):
+            abs_path_file = dir_path / file
+            direction_string = ".+_R" + str(read_direction) + "_[0-9]+.fastq.gz"
+            direction_pattern = re.compile(direction_string)
+            if direction_pattern.match(str(abs_path_file)):
+                same_direction.append(os.path.abspath(abs_path_file))
+        same_direction.sort()
+        return same_direction
+
+    def get_total_size(self, files: list) -> int:
+        total_size = 0
+        for file in files:
+            total_size = total_size + Path(file).stat().st_size
+        return total_size
+
+    def concatenate_same_read_direction(self, reads: list, output: Path) -> None:
+        with open(output, "wb") as write_file_obj:
+            for file in reads:
+                with open(file, "rb") as file_descriptor:
+                    shutil.copyfileobj(file_descriptor, write_file_obj)
+
+    def remove_files_with_less_than_one_inode(self, reads: list) -> None:
+        for file in reads:
+            inode_check_cmd = "stat -c %h " + file
+            n_inodes = subprocess.getoutput(inode_check_cmd)
+            if int(n_inodes) > 1:
+                LOG.info("Removing file: %s" % (file))
+                os.remove(file)
+            else:
+                LOG.warning("WARNING %s only got 1 inode, file will not be removed" % (file))
+
     def concatenate(self, ticket_id: int, dry_run: bool) -> None:
-        customer_inbox = self.get_inbox_path(ticket_id=ticket_id)
-        for dir_name in os.listdir(customer_inbox):
-            dir_path = os.path.join(customer_inbox, dir_name)
+        customer_inbox: Path = self.get_inbox_path(ticket_id=ticket_id)
+        if not customer_inbox.exists() and dry_run:
+            LOG.info("Dry run, nothing will be concatenated in: %s", customer_inbox)
+            return
+        if not customer_inbox.exists():
+            LOG.info("The path %s do not exist, nothing will be concatenated", customer_inbox)
+            return
+        for dir_name in customer_inbox.iterdir():
+            dir_path = customer_inbox / dir_name
```

Comment:
```suggestion
            dir_path: Path = customer_inbox / dir_name
```

Reply:
I think there are some confusion around `dir_name` and `dir_path` here. Since `dir_name` are `Path` objects they already include the information of the rest of the path...